docs/usage/debugging.rst

```diff
@@ -0,0 +1,116 @@
+Debugging
+=========
+
+Using the Python debugger
+--------------------------
+
+You can configure Litestar to drop into the :doc:`Python Debugger <python:library/pdb>`
+when an exception occurs. This can be configured in different ways:
+
+Configuring ``Litestar`` with the ``pdb_on_exception`` option
+    .. code-block:: python
+
+        app = Litestar(pdb_on_exception=True)
+
+
+Running your app with the CLI and using the ``--pdb`` flag
+    .. code-block:: shell
+
+        litestar run --pdb
+
+Using the ``LITESTAR_PDB`` environment variable
+    ``LITESTAR_PDB=1``
+
+
+Debugging with an IDE
+---------------------
+
+You can easily attach your IDEs debugger to your application, whether you're running it
+via the CLI or a webserver like `uvicorn <https://www.uvicorn.org/>`_.
+
+Intellij / PyCharm
+++++++++++++++++++
+
+Using the CLI
+*************
+
+1. Create a new debug configuration via ``Run`` > ``Edit Configurations``
+2. Select ``Module name`` option and set it to ``litestar``
+3. Add the ``run`` parameter and optionally additional parameters you want to pass
+   to the CLI
+
+   .. image:: /images/debugging/pycharm-config-cli.png
```

Comment:
I had to set the working directory to get this to work for some reason:

![image](https://github.com/litestar-org/litestar/assets/20659309/ed0b6432-c540-42b8-8688-9dcd42d47d80)

Without it:

![image](https://github.com/litestar-org/litestar/assets/20659309/bf7761c0-6666-4941-b832-77579c2c2571)

Reply:
> I had to set the working directory to get this to work for some reason:

That's needed. I have this in my template though and forgot. Will fix.